src/lighteval/models/abstract_model.py

```diff
@@ -183,23 +183,20 @@ def tok_encode_pair(self, context, continuation, pairwise: bool = False):
             context = context[:-n_spaces]
 
         if pairwise:
-            context_enc, continuation_enc = self.tok_encode(context), self.tok_encode(continuation)
-            if self.add_special_tokens:
-                tokenized_with_special_tokens = self.tokenizer.build_inputs_with_special_tokens(
-                    context_enc + continuation_enc
-                )
-                # If this fails something went wrong as the function above should only add special tokens
-                first_non_prefix_token_idx = tokenized_with_special_tokens.index(context_enc[0])
-                last_context_token_idx = first_non_prefix_token_idx + len(context_enc)
-                context_enc, continuation_enc = (
-                    tokenized_with_special_tokens[:last_context_token_idx],
-                    tokenized_with_special_tokens[last_context_token_idx:],
-                )
+            context_enc, continuation_enc = (
+                self.tok_encode(context, add_special_tokens=self.add_special_tokens),
+                self.tok_encode(continuation, add_special_tokens=self.add_special_tokens),
```

Comment:
1. Did you make sure that 
```
self.tokenizer(
            str_to_encode,
            padding=True,
            add_special_tokens=add_special_tokens,
            return_tensors="pt",
        )
```
is indeed equivalent to `self.tokenizer.build_inputs_with_special_tokens`?

2. You're removing the catch below, not sure all models will still pass (cc @NathanHB who added it iirc, do you have a model for which you remember this was needed?)

Reply:
I was the one who implemented this. The issue here is that `tokenizer.build_inputs_with_special_tokens` doesn't work for models that don't inherit (and override build_inputs_with_special_tokens) the PretrainedTokenizer.

This is a case of LLamaTokenizer, which will not work. `tokenizer.build_inputs_with_special_tokens` and will not add BOS in this case, while tokenizer.encode will.

Also note that even without this simplification the method was not implemented correctly:
```
context_enc, continuation_enc = self.tok_encode(context), self.tok_encode(continuation)
```
Was used, which meant that the special tokens were even before build_inputs_with_sepcial_tokens.
This meant that the continuation could have also received the special tokens.

I noticed this when I was using llama3.2 3B for mmlu testing and found out that it was adding bos in front of continuation. This made the scores "tank" and be around random baseline. After this fix the score were fine.

Reply:
OK, then that solves it on my side :sweat_smile: